src/DynamicData/List/Internal/TransformAsync.cs

```diff
@@ -88,25 +97,44 @@ private async Task Transform(ChangeAwareList<TransformedItemContainer> transform
 
                 case ListChangeReason.AddRange:
                     {
-                        var tasks = item.Range.Select(_containerFactory);
+                        var startIndex = item.Range.Index < 0 ? transformed.Count : item.Range.Index;
+                        var tasks = item.Range.Select((t, idx) => _containerFactory(t, Optional<TDestination>.None, idx + startIndex));
                         var containers = await Task.WhenAll(tasks).ConfigureAwait(false);
                         transformed.AddOrInsertRange(containers, item.Range.Index);
                         break;
                     }
 
+                case ListChangeReason.Refresh:
+                    {
+                        var change = item.Item;
+                        if (_transformOnRefresh)
+                        {
+                            Optional<TDestination> previous = transformed[change.CurrentIndex].Destination;
+                            var container = await _containerFactory(change.Current, previous, change.CurrentIndex)
+                                .ConfigureAwait(false);
+                            transformed.Refresh(container, change.CurrentIndex);
```

Comment:
`transformed.Refresh(container, change.CurrentIndex); `

should be 

` transformed[change.CurrentIndex] = container;`

because technically transform on refresh is an instruction that the we we need to re-apply a transform which creates (or normally creates) a new item.

On my PC, this fixes the  List.TransformAsyncFixture.TransformOnRefresh test

Reply:
👍